pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstore_test.go

```diff
@@ -0,0 +1,1827 @@
+			},
+			want:    false,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sigstore := &sigstoreImpl{
+				verifyFunction:             tt.fields.verifyFunction,
+				skippedImages:              tt.fields.skippedImages,
+				fetchImageManifestFunction: tt.fields.fetchImageManifestFunction,
+			}
+			got, err := sigstore.ValidateImage(tt.args.ref)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("sigstoreImpl.ValidateImage() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("sigstoreImpl.ValidateImage() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestSigstoreimpl_AddAllowedSubject(t *testing.T) {
+	type fields struct {
+		subjectAllowList map[string]bool
+	}
+	type args struct {
+		subject string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   map[string]bool
+	}{
+		{
+			name: "add allowed subject to nil map",
+			fields: fields{
+				subjectAllowList: nil,
+			},
+			args: args{
+				subject: "spirex@example.com",
+			},
+			want: map[string]bool{
+				"spirex@example.com": true,
+			},
+		},
+		{
+			name: "add allowed subject to empty map",
+			fields: fields{
+				subjectAllowList: map[string]bool{},
+			},
+			args: args{
+				subject: "spirex@example.com",
+			},
+			want: map[string]bool{
+				"spirex@example.com": true,
+			},
+		},
+		{
+			name: "add allowed subject to existing map",
+			fields: fields{
+				subjectAllowList: map[string]bool{
+					"spirex1@example.com": true,
+					"spirex2@example.com": true,
+					"spirex3@example.com": true,
+					"spirex5@example.com": true,
+				},
+			},
+			args: args{
+				subject: "spirex4@example.com",
+			},
+			want: map[string]bool{
+				"spirex1@example.com": true,
+				"spirex2@example.com": true,
+				"spirex3@example.com": true,
+				"spirex4@example.com": true,
+				"spirex5@example.com": true,
+			},
+		},
+		{
+			name: "add existing allowed subject to existing map",
+			fields: fields{
+				subjectAllowList: map[string]bool{
+					"spirex1@example.com": true,
+					"spirex2@example.com": true,
+					"spirex3@example.com": true,
+					"spirex4@example.com": true,
+					"spirex5@example.com": true,
+				},
+			},
+			args: args{
+				subject: "spirex4@example.com",
+			},
+			want: map[string]bool{
+				"spirex1@example.com": true,
+				"spirex2@example.com": true,
+				"spirex3@example.com": true,
+				"spirex4@example.com": true,
+				"spirex5@example.com": true,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sigstore := &sigstoreImpl{
+				subjectAllowList: tt.fields.subjectAllowList,
+			}
+			sigstore.AddAllowedSubject(tt.args.subject)
+			if !reflect.DeepEqual(sigstore.subjectAllowList, tt.want) {
+				t.Errorf("sigstore.subjectAllowList = %v, want %v", sigstore.subjectAllowList, tt.want)
+			}
+		})
+	}
+}
+
+func TestSigstoreimpl_ClearAllowedSubjects(t *testing.T) {
```

Comment:
same comment for all tests that has a single test case.

It is an overkill to use test table driving for a single test case, you must use a single t.Run instead